client/lib/protect-form/index.jsx

```diff
@@ -57,6 +67,41 @@ export const protectForm = WrappedComponent => {
 	};
 };
 
+/*
+ * Declarative variant that takes a 'isChanged' prop.
+ */
+export class ProtectForm extends Component {
```

Comment:
> Longer term, do we plan to keep both?

I don't know at this moment. The declarative component works much better with Redux, but the HOC one may remain useful for some cases forever.

Anyway, both components will have to live next to each other for some time, so I renamed the declarative one to `ProtectFormGuard` to avoid naming confusion.